src/test/shell/bazel/remote/build_without_the_bytes_test.sh

```diff
@@ -221,12 +221,54 @@ EOF
     --remote_download_toplevel \
     //a:foobar >& $TEST_log || fail "Failed to build //a:foobar"
 
-  expect_log "2 processes: 1 remote cache hit, 1 internal"
+  expect_log "1 process: 1 internal"
```

Comment:
> Interesting, so we won't report a fetch as a cache hit unless the action actually ran?

This is a little bit complex.

Before this change, we decide whether to download outputs of a spawn by checking whether any of its output is included in the set `filesToDownload` (the set is computed in the `afterAnalysis` callback in the `RemoteModule` based on analysis result). If we decided to download the outputs of the spawn, we download all the outputs but **do not** inject the metadata. It means we let skyframe to track the outputs on the local filesystem. So if we delete the output, skyframe will rerun the generating action, in which case, it will hit the remote cache and re-download the output.

With this change, we **always** inject metadata for outputs even if we decide to download them later. It means skyframe will use the metadata to track outputs. If we delete the local outputs, the metadata is still there, in which case, it will hit the skyframe cache.